core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h

```diff
@@ -321,22 +411,284 @@ STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_multiply(
     mp_raise_ValueError("Invalid public key");
   }
   uint8_t out[65];
+  size_t outlen = sizeof(out);
   if (!secp256k1_ecdh(ctx, out, &ec_pk, (const uint8_t *)sk.buf,
                       secp256k1_ecdh_hash_passthrough, NULL)) {
     mp_raise_ValueError("Multiply failed");
   }
+  if (compressed_result) {
+    // EC pubkey is big-endian, so we test the least significant byte of y.
+    out[0] = (out[outlen - 1] & 1) ? SECP256K1_TAG_PUBKEY_ODD
+                                   : SECP256K1_TAG_PUBKEY_EVEN;
+    outlen = 33;
+  }
+  return mp_obj_new_bytes(out, outlen);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(
+    mod_trezorcrypto_secp256k1_context_multiply_obj, 3, 4,
+    mod_trezorcrypto_secp256k1_context_multiply);
+
+/// def blind_generator(self, asset: bytes, blinder: bytes) -> bytes:
+///     '''
+///     Generate blinded generator for the specified confidential asset.
+///     '''
+STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_blind_generator(
+    mp_obj_t self, mp_obj_t asset_obj, mp_obj_t blind_obj) {
+  const secp256k1_context *ctx = mod_trezorcrypto_get_secp256k1_context(self);
+
+  mp_buffer_info_t asset;
+  mp_get_buffer_raise(asset_obj, &asset, MP_BUFFER_READ);
+  if (asset.len != 32) {
+    mp_raise_ValueError("Invalid length of asset");
+  }
+  mp_buffer_info_t blind;
+  mp_get_buffer_raise(blind_obj, &blind, MP_BUFFER_READ);
+  if (blind.len != 32) {
+    mp_raise_ValueError("Invalid length of blinding factor");
+  }
+  secp256k1_generator gen;
+  if (!secp256k1_generator_generate_blinded(ctx, &gen, asset.buf, blind.buf)) {
+    mp_raise_ValueError("Generator blinding failed");
+  }
+  byte out[33] = {0};
+  secp256k1_generator_serialize(ctx, out, &gen);
```

Comment:
nit: can't hurt to assert the return value here.

Reply:
Added an assertion.